pypika/terms.py

```diff
@@ -335,25 +337,27 @@ def __init__(self, value: Any, alias: Optional[str] = None) -> None:
         super().__init__(alias)
         self.value = value
 
-    def get_value_sql(self, **kwargs: Any) -> str:
+    def get_value_sql(self, _value: Any = UNSET, **kwargs: Any) -> str:
         quote_char = kwargs.get("secondary_quote_char") or ""
 
+        # allow `get_value_sql` to be called recursively
+        value = self.value if _value is UNSET else _value
```

Comment:
That's honestly very hacky, to not say poor OOP design, and definitely adds a lot of complexity to this method. Adding a param in this way, to be overriding the default behaviour of this method is something I find really hard to justify.

Reply:
PR already updated. I agree this solution is a little hacky